spec/ruby/core/numeric/shared/step.rb

```diff
@@ -0,0 +1,397 @@
+  end
+
+  it "returns an Enumerator when step is 0" do
+    1.send(@method, *get_args(@args_type, 2, 0)).should be_an_instance_of(enumerator_class)
+  end
+
+  it "returns an Enumerator when not passed a block and self > stop" do
+    1.send(@method, *get_args(@args_type, 0, 2)).should be_an_instance_of(enumerator_class)
+  end
+
+  it "returns an Enumerator when not passed a block and self < stop" do
+    1.send(@method, *get_args(@args_type, 2, 3)).should be_an_instance_of(enumerator_class)
+  end
+
+  it "returns an Enumerator that uses the given step" do
+    0.send(@method, *get_args(@args_type, 5, 2)).to_a.should == [0, 2, 4]
+  end
+
+  describe "with [stop, step]" do
+    before :each do
+      @stop = mock("Numeric#step stop value")
+      @step = mock("Numeric#step step value")
+      @obj = NumericSpecs::Subclass.new
+    end
+
+    it "increments self using #+ until self > stop when step > 0" do
+      @step.should_receive(:>).with(0).and_return(true)
+      @obj.should_receive(:>).with(@stop).and_return(false, false, false, true)
+      @obj.should_receive(:+).with(@step).and_return(@obj, @obj, @obj)
+
+      @obj.send(@method, *get_args(@args_type, @stop, @step), &@prc)
+
+      ScratchPad.recorded.should == [@obj, @obj, @obj]
+    end
+
+    it "decrements self using #+ until self < stop when step < 0" do
+      @step.should_receive(:>).with(0).and_return(false)
+      @obj.should_receive(:<).with(@stop).and_return(false, false, false, true)
+      @obj.should_receive(:+).with(@step).and_return(@obj, @obj, @obj)
+
+      @obj.send(@method, *get_args(@args_type, @stop, @step), &@prc)
+
+      ScratchPad.recorded.should == [@obj, @obj, @obj]
+    end
+  end
+
+  describe "Numeric#step with [stop, step] when self, stop and step are Fixnums" do
+    it "yields only Fixnums" do
+      1.send(@method, *get_args(@args_type, 5, 1)) { |x| x.should be_kind_of(Fixnum) }
+    end
+  end
+
+  describe "Numeric#step with [stop, step] when self and stop are Fixnums but step is a String" do
+    it "returns an Enumerator if not given a block" do
+      1.send(@method, *get_args(@args_type, 5, "1")).should be_an_instance_of(enumerator_class)
+      1.send(@method, *get_args(@args_type, 5, "0.1")).should be_an_instance_of(enumerator_class)
+      1.send(@method, *get_args(@args_type, 5, "1/3")).should be_an_instance_of(enumerator_class)
+      1.send(@method, *get_args(@args_type, 5, "foo")).should be_an_instance_of(enumerator_class)
+    end
+
+    it "raises an ArgumentError if given a block" do
+      lambda { 1.send(@method, *get_args(@args_type, 5, "1")) {} }.should raise_error(ArgumentError)
+      lambda { 1.send(@method, *get_args(@args_type, 5, "0.1")) {} }.should raise_error(ArgumentError)
+      lambda { 1.send(@method, *get_args(@args_type, 5, "1/3")) {} }.should raise_error(ArgumentError)
+      lambda { 1.send(@method, *get_args(@args_type, 5, "foo")) {} }.should raise_error(ArgumentError)
```

Comment:
All these expectations should be moved into separate `it` blocks, same goes for the specs below. For example:

```ruby
describe 'with a block' do
  it 'raises ArgumentError with step "0.1"' do
    
  end

  it 'raises ArgumentError with step "1/3"' do
    
  end

  ...
end
```